### PR TITLE
refactor(e2e): simplify playwright test: landing.spec.ts

### PR DESCRIPTION
**Pull request by anand-harsh**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

#51705

### CODE REVIEW

**Comment by anand-harsh** [comment]
Tests are passing in local but failing here. Why is it so?

File: e2e/landing.spec.ts [hunk 1]
```diff
@@ -82,29 +99,18 @@ test('The as seen in container is visible with featured logos', async () => {
...
-    ).toBeVisible();
-    await expect(
-      card.getByTestId('testimonials-endorser-occupation')
-    ).toBeVisible();
-    await expect(
-      card.getByTestId('testimonials-endorser-testimony')
-    ).toBeVisible();
+    for (let i = 0; i < testimonialEndorser.length; i++) {
+      await expect(card.getByTestId(testimonialEndorser[i])).toBeVisible();
+    }
```
[the remark this answers is not in the document]
Comment (anand-harsh) [reply e2e/landing.spec.ts]: I have updated the PR, can you please check if the changes are ok and suggest if some more needed. I can't replace it with getByText as there are card components like pieces and components of each card has been taken by getByTestId here.

**Comment by anand-harsh** [comment]
@huyenltnguyen I have updated the PR, can you please check if the changes are ok and suggest if some more needed. I can't replace it with getByText as there are card components like pieces and components of each card has been taken by getByTestId here.

**Review by huyenltnguyen — APPROVED** [review]

File: e2e/landing.spec.ts [hunk 2]
```diff
@@ -111,9 +115,9 @@ test('Testimonial endorser people have images, occupation, location and testimon
 test('Has links to all superblocks', async () => {
   const curriculumBtns = page.getByTestId(landingPageElements.curriculumBtns);
   await expect(curriculumBtns).toHaveCount(15);
-  for (let i = 0; i < superBlocks.length; i++) {
-    const btn = curriculumBtns.nth(i);
-    await expect(btn).toContainText(superBlocks[i]);
+  for (let item = 0; item < superBlocks.length; item++) {
+    const btn = curriculumBtns.nth(item);
+    await expect(btn).toContainText(superBlocks[item]);
```
Comment (naomi-lgbt) [inline e2e/landing.spec.ts:120]:
```suggestion
  for (let index = 0; index < superBlocks.length; index++) {
    const btn = curriculumBtns.nth(index);
    await expect(btn).toContainText(superBlocks[index]);
```

Nitpick, but if we're going to deviate from the standard `i` for loop iterators, we should at least name it correctly.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**